source/MaterialXCore/Definition.cpp

```diff
@@ -36,22 +36,34 @@ InterfaceElementPtr NodeDef::getImplementation(const string& target, const strin
     vector<InterfaceElementPtr> interfaces = getDocument()->getMatchingImplementations(getQualifiedName(getName()));
     vector<InterfaceElementPtr> secondary = getDocument()->getMatchingImplementations(getName());
     interfaces.insert(interfaces.end(), secondary.begin(), secondary.end());
+
+    // Scan for implementations, looking for those with a language match 
+    // over those which do not.
+    bool matchLanguage = !language.empty();
     for (InterfaceElementPtr interface : interfaces)
     {
-        if (!targetStringsMatch(interface->getTarget(), target) ||
+        ImplementationPtr implement = interface->asA<Implementation>();
+        if (!implement||
+            !targetStringsMatch(interface->getTarget(), target) ||
             !isVersionCompatible(interface))
         {
             continue;
         }
-        if (!language.empty())
+        if (!matchLanguage || 
```

Comment:
First scan for implementations. If no language is specified then it will just take the first implementation as it did before at line 56 in the previous code.